### PR TITLE
C# - add std::list

### PR DESCRIPTION
**Pull request by Liryna**
This continue the work of @schullq https://github.com/swig/swig/pull/989

@wsfulton, I have renable the `==` operator. Is everything alright now ?

### CODE REVIEW

**Comment by ojwb** [comment]
The CI failures are for python, and seem unrelated to the changes here.

File: Examples/test-suite/li_std_list.i [hunk 1]
```diff
@@ -38,13 +38,23 @@ struct Struct {
   double num;
   Struct() : num(0.0) {}
   Struct(double d) : num(d) {}
-//  bool operator==(const Struct &other) { return (num == other.num); }
+  bool operator==(const Struct &other) { return (num == other.num); }
```
Comment (wsfulton) [inline Examples/test-suite/li_std_list.i]: Please revert this change. The `operator==` must remain commented out and the appropriate changes made to deal with it missing. This was mentioned in #989 and is because most classes do not have an `operator==` and std::list does not require the container type to have it.

Comment (wsfulton) [reply Examples/test-suite/li_std_list.i]: I've put in the required changes in 78051fdd33b5dc5fca0ec5787821ce5f9030af93

**Comment by wsfulton** [comment]
Merged in e7f9316e6037c97d005abd97cf7d12bb2582ccc4
